modules/microshift-config-snippets.adoc

```diff
@@ -0,0 +1,117 @@
+//Q: Per current docs, there is no `config.yaml` unless the user creates one. Is a user-created config.yaml a prerequisite?
+
+//To create custom configurations, make a copy of the `config.yaml.default` file that is provided in the `/etc/microshift/` directory, renaming it `config.yaml`. Keep this file in the `/etc/microshift/` directory, and then you can change supported settings that are expected to override the defaults before starting or restarting MicroShift.
+
+Files in the snippet directory, `/etc/microshift/config.d`, are sorted in lexicographical order and run sequentially. You can use numerical prefixes for snippets so that each is read in the order you want.
+
+Values in an existing {microshift-short} configuration file are overwritten at startup and the snippets are automatically incorporated into the `config.yaml` file.
+//Q: So what happens if I am the new admin and I haven't paid attention to the `config.d` directory. I change something directly in the `config.yaml` but the snippets are still there. Do they overwrite my `config.yaml` settings when I restart MicroShift?
+
+.{microshift-short} configuration directory contents
+* `/etc/microshift/config.yaml`
+
+.Example {microshift-short} configuration snippet directory contents
+* `/etc/microshift/config.d/10-subjectAltNames.yaml`
+* `/etc/microshift/config.d/20-kubelet.yaml`
+
+Lists are not merged, they are overwritten. For example, you can replace a SAN or list of SANs by creating an additional snippet for the same field that is read after the first:
+
+.{microshift-short} configuration directory contents
+* `/etc/microshift/config.yaml`
+
+.Example {microshift-short} configuration snippet directory contents
+* `/etc/microshift/config.d/10-san.yaml`
+* `/etc/microshift/config.d/20-san.yaml`
+
+.Example `10-san.yaml` snippet
+[source,yaml]
+----
+apiServer:
+  subjectAltNames:
+    - host1
+    - host2
+----
+
+.Example `20-san.yaml` snippet
+[source,yaml]
+----
+apiServer:
+  subjectAltNames:
+    - hostZ
+----
+
+.Example `config.yaml` result
+[source,yaml]
+----
+apiServer:
+# ---
+  subjectAltNames:
+    - hostZ
+# ---
+----
+
+If you want to add a value to an existing list, you can add it to an existing snippet. For example, to add `hostZ` to an existing list of SANs, add it to your snippet:
+
+.Example `10-san.yaml` snippet
+[source,yaml]
+----
+apiServer:
+  subjectAltNames:
+    - host1
+    - host2
+    - hostZ
+----
+
+.Example `config,yaml` result
+[source,yaml]
+----
+apiServer:
+# ---
+  subjectAltNames:
+    - host1
+    - host2
+    - hostZ
+# ---
+----
+
+Contents of kubelet configurations are merged together, unless the specific field is a list.
+//Is this only true for every field that isn't a list? For example, can I add an IPv6 address (or change my storage driver and CSI when supported) with a snippet?
```

Comment:
Yes. Looking from perspective of JSON: objects [1] are merged, arrays [2] are overwritten.
1: `{"a": 1} + {"b": 2} => {"a":1, "b":2}`
2: `{"a": [1,2,3]} + {"a": [4,5,6]} => {"a": [4,5,6]}`